### ISSUE TITLE
Remove cyclic dependency between dwio_common and dwio_common_compression

### ISSUE THREAD

**Issue by assignUser** [facebookincubator/velox#6128]
### Bug description

#5946 introduced a real  cyclic dependency between dwio_common and dwio_common_compression:
```
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "velox_dwio_common" of type SHARED_LIBRARY
    depends on "velox_dwio_common_compression" (weak)
  "velox_dwio_common_compression" of type SHARED_LIBRARY
    depends on "velox_dwio_common" (weak)
```

This should  be removed as it is bad practice and blocks the building of shared libraries.

### System information

-

**Comment by nmahadevuni** [comment]
This code was present prior to #5946 . Both these files are not modified by this PR. The header file you have given above is part of velox_common_compression and not velox_dwio_common_compression.

**Comment by assignUser** [comment]
Ah then [this addition](https://github.com/facebookincubator/velox/pull/5946/files#diff-21609ca7e74a6cc3d1dca0a477db83f1ece280d234e81519f7506e05d3a17d9dR65) was potentially superfluous and creating the cycle? (haven't had time to check a full build)
```
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "velox_dwio_common" of type SHARED_LIBRARY
    depends on "velox_dwio_common_compression" (weak)
  "velox_dwio_common_compression" of type SHARED_LIBRARY
    depends on "velox_dwio_common" (weak)
```

**Comment by czentgr** [comment]
@assignUser I suppose this can be closed.

**Comment by assignUser** [comment]
Fixed by #6760
